var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -123,6 +124,8 @@ class NetcdfC(AutotoolsPackage):
     # https://github.com/Unidata/netcdf-c/issues/250
     depends_on("hdf5@:1.8", when="@:4.4.0")
 
+    depends_on("zstd", when="@4.9.0:")
```

Comment:
As I have already mentioned, I would like to have `zstd` as an optional dependency. It might be the case that we don't even have to patch the configure script for that and just simply set a cache variable on the command line like we do for [`curl`](https://github.com/spack/spack/blob/42de3ff968f8bc32221ced071ba3633bca670bb4/var/spack/repos/builtin/packages/netcdf-c/package.py#L187).

Reply:
And before I forget. There is a caveat when declaring variants with `when`: if `+var1 in self.spec` is `False`, it does not mean that `~var1 in self.spec` is `True` because versions that do not fall into `when` of `var1` have `False` in both cases. So, it might be the case that a clean solution for `zstd` would be something like:
```python
if "+zstd" in self.spec:
    # append to cppflags and ldflags
elif "~zstd" in self.spec:
    # prevent linking to a system installation of zstd
```

Reply:
Done, setting the variable `ac_cv_lib_zstd_ZSTD_compress=no` worked.